pkg/comp-functions/functions/vshnmariadb/mariadb_deploy.go

```diff
@@ -0,0 +1,263 @@
+			"enabled": true,
+			"containerSecurityContext": map[string]interface{}{
+				"enabled": false,
+			},
+		},
+		"securityContext": map[string]interface{}{
+			"enabled": false,
+		},
+		"containerSecurityContext": map[string]interface{}{
+			"enabled": false,
+		},
+		"podSecurityContext": map[string]interface{}{
+			"enabled": false,
+		},
+	}
+
+	vb, err := json.Marshal(values)
+	if err != nil {
+		err = fmt.Errorf("cannot marshal helm values: %w", err)
+		return err
+	}
+
+	r := &xhelmbeta1.Release{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: comp.GetName(),
+		},
+		Spec: xhelmbeta1.ReleaseSpec{
+			ForProvider: xhelmbeta1.ReleaseParameters{
+				Chart: xhelmbeta1.ChartSpec{
+					Repository: svc.Config.Data["chartRepository"],
+					Version:    svc.Config.Data["chartVersion"],
+					Name:       "mariadb-galera",
+				},
+				Namespace: comp.GetInstanceNamespace(),
+				ValuesSpec: xhelmbeta1.ValuesSpec{
+					Values: k8sruntime.RawExtension{
+						Raw: vb,
+					},
+				},
+			},
+			ResourceSpec: xpv1.ResourceSpec{
+				ProviderConfigReference: &xpv1.Reference{
+					Name: "helm",
+				},
+				WriteConnectionSecretToReference: &xpv1.SecretReference{
+					Name:      comp.GetName() + "-connection",
+					Namespace: comp.GetInstanceNamespace(),
+				},
+			},
+			ConnectionDetails: []xhelmbeta1.ConnectionDetail{
+				{
+					ObjectReference: corev1.ObjectReference{
+						APIVersion: "v1",
+						Kind:       "Secret",
+						Name:       comp.GetName(),
+						Namespace:  comp.GetInstanceNamespace(),
+						FieldPath:  "data.mariadb-root-password",
+					},
+					ToConnectionSecretKey: "MARIADB_PASSWORD",
+				},
+				{
+					ObjectReference: corev1.ObjectReference{
+						APIVersion: "v1",
+						Kind:       "Service",
+						Name:       comp.GetName(),
+						Namespace:  comp.GetInstanceNamespace(),
+						FieldPath:  "spec.ports[0].port",
+					},
+					ToConnectionSecretKey: "MARIADB_PORT",
+				},
+				{
+					ObjectReference: corev1.ObjectReference{
+						APIVersion: "v1",
+						Kind:       "Secret",
+						Name:       "tls-server-certificate",
+						Namespace:  comp.GetInstanceNamespace(),
+						FieldPath:  "data[ca.crt]",
+					},
+					ToConnectionSecretKey:  "ca.crt",
+					SkipPartOfReleaseCheck: true,
+				},
+			},
+		},
+	}
+
+	err = svc.AddObservedConnectionDetails(comp.Name + "-release")
+	if err != nil {
+		return err
+	}
+
+	return svc.SetDesiredComposedResourceWithName(r, comp.Name+"-release")
+}
+
+func getConnectionDetails(ctx context.Context, comp *vshnv1.VSHNMariaDB, svc *runtime.ServiceRuntime) error {
+	mariadbHost := comp.GetName() + ".vshn-mariadb-" + comp.GetName() + ".svc.cluster.local"
+	mariadbURL := fmt.Sprintf("mysql://%s:%s", mariadbHost, svc.GetConnectionDetails()["MARIADB_PORT"])
+
+	svc.SetConnectionDetail("MARIADB_HOST", []byte(mariadbHost))
+	svc.SetConnectionDetail("MARIADB_USERNAME", []byte("root"))
+	svc.SetConnectionDetail("MARIADB_URL", []byte(mariadbURL))
+
+	return nil
```

Comment:
Did you notice that you don't actually have a single error in this function? You can remove this, if you go the fixed port route.

Otherwise, please check if the `svc.GetConnectionDetails()["MARIADB_PORT"]` entry actually exists or not and produce an error accordingly.

Reply:
Removed the return value as it's not needed anymore